components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -160,7 +162,15 @@
                 identityMgtEndpointContext = IdentityUtil.getServerURL("/accountrecoveryendpoint", true, true);
             }
 
-            String url = identityMgtEndpointContext + "/recoverpassword.do?callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+            HttpClient client = IdentityUtil.getHttpClientWithSSLSocketFactory(scheme, serverName);
```

Comment:
Can we check the servlet context registered or not here, instead of executing a head call?